io/pci/distro_tools.py

```diff
@@ -74,19 +76,69 @@ def lsprop(self, option):
         if result.exit_status != 0:
             self.fail("lsprop failed")
 
+    def lsvio(self, option):
+        '''
+        run lsvio aaplicable only for PowerVM
+        '''
+        cmd = "lsvio -%s" % option
+        result = process.run(cmd, shell=True, ignore_status=True)
+        if result.exit_status != 0:
+            self.fail("lsvio failed")
+
+    def lsdevinfo(self, option):
+        '''
+        run lsdevinfo
+        '''
+        cmd = "lsdevinfo -%s" % option
+        result = process.run(cmd, shell=True, ignore_status=True)
+        if result.exit_status != 0:
+            self.fail("lsdevinfo failed")
+
+    def usys(self, tool, option, pci_device):
+        '''
+        run usysident and usysattn
+        '''
+        location_code = pci.get_slot_from_sysfs(pci_device)
+        interface = pci.get_interfaces_in_pci_address(pci_device, "net")[0]
+        cmd = "%s %s" % (tool, option)
+        if '-P' in cmd:
+            cmd += " -l %s" % location_code
+        if '-t' in cmd:
+            cmd += " -d %s" % interface
+        result = process.run(cmd, shell=True, ignore_status=True)
+        if tool == "usysident":
+            if "There is no identify indicator" in result.stdout_text:
+                self.log.warn("%s option %s failed" % (tool, option))
+        elif tool == "usysattn":
+            if "There is no fault indicator" in result.stdout_text:
+                self.log.warn("%s option %s failed" % (tool, option))
+
+    def usysattn(self, option, pci_device):
+        self.usys(self.tool, option, pci_device)
+        return
+
+    def usysident(self, option, pci_device):
+        self.usys(self.tool, option, pci_device)
+        return
+
+    def ofpathname(self, option, pci_device):
+        '''
+        run ofpathname aaplicable only for PowerVM
+        '''
+        interface = pci.get_interfaces_in_pci_address(pci_device, "net")[0]
+        cmd = "ofpathname -%s %s" % (option, interface)
+        result = process.run(cmd, shell=True, ignore_status=True)
+        if result.exit_status != 0:
+            self.fail("ofpathname failed")
+
     def test(self):
         '''
         test different distro tools
         '''
-        if self.tool == "lsslot":
-            for val in self.option:
-                self.lsslot(val)
-        elif self.tool == "netstat":
-            for val in self.option:
-                self.netstat(val)
-        elif self.tool == "lsprop":
-            for val in self.option:
-                self.lsprop(val)
+        if self.tool in ('lsslot', 'netstat', 'lsprop', 'lsvio', 'lsdevinfo'):
+            getattr(self, self.tool)(self.option)
+        else:
+            getattr(self, self.tool)(self.option, self.pci_device)
```

Comment:
@smruti77 Here, since option and pci_device can be handled inside those functions, we need not pass them. The functions can directly access them using a self

Reply:
@smruti77 can this be taken care please ?